x-pack/platform/plugins/shared/sample_data_ingest/server/routes/installation.ts

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IRouter } from '@kbn/core/server';
+import { i18n } from '@kbn/i18n';
+import {
+  INSTALL_API_PATH,
+  STATUS_API_PATH,
+  DatasetSampleType,
+  type StatusResponse,
+  type InstallResponse,
+} from '../../common';
+import { InternalServices } from '../types';
+
+export const registerInstallationRoutes = ({
+  router,
+  getServices,
+}: {
+  router: IRouter;
+  getServices: () => InternalServices;
+}) => {
+  router.post(
+    {
+      path: INSTALL_API_PATH,
+      security: {
+        authz: {
+          enabled: false,
+          reason: 'This route delegates authorization to the scoped ES client',
+        },
+      },
+      validate: false,
+      options: {
+        access: 'internal',
+        timeout: { idleSocket: 2 * 60 * 1000 }, // install can take time.
```

Comment:
Theres a status endpoint right? can we start the install then return `201` ACCEPTED. to not block the request? If this can take minutes we should try to make this async. Granted that may have complications I'm not aware of.

Reply:
Yeah I spoke with Sander about running this task in taskManager so we would have async install. Decided to keep it like that for now, and then improve